### PR TITLE
Display and load sc6 files in "load game" browser

### PR DESCRIPTION
**Pull request by wolfreak99**
This makes use of string tokens and the "scenario_load_and_play_from_file" that is used in the scenario select window to extend the files shown in the load game browser. Because sv6 and sc6 files are seperate things, I chose to show extensions for the sc6 extension. this may also allow for showing sv4 and sc4 files in the browser as well, but i am unsure if those will load properly as is. I'm also not sure if loading a scenario this way will work when running servers, now that i think of the callback function in the scenarioselect window.... I'm about to test that.

EDIT: This will not work when running a server. I'm looking to fix this.
EDIT2: Turns out it not working while running a server wasn't because of this, as attempting to load a game on network in vanilla OpenRCT2 also triggers the crash. It is something to do with comparing a vector somewhere, I am still looking to find the issue and to fix it.
EDIT3: I have found the underlying issue with the bug mentioned above and have posted a seperate PR to address that (#4780). This can now be reviewed and merged more comfortably.

### CODE REVIEW

**Comment by Gymnasiast** [comment]
While you're at it, could you extend it to .SV4 as well?

**Comment by wolfreak99** [comment]
@Gymnasiast :) Hell yeah. Theres a problem though - I don't have any SV4 files to test it with.. would you be able to share a test SV4 and SC4?

**Comment by wolfreak99** [comment]
@Gymnasiast there ya go.. With the exception of td4s (but the code seems to already be prepared for files of td4), every file format has been tested

File: src/scenario.c [hunk 1]
```diff
@@ -129,7 +130,16 @@ int scenario_load_and_play_from_path(const char *path)
...
+	if (extension == FILE_EXTENSION_SC6) {
+		if (!scenario_load(path))
+			return 0;
+	}
+	else if (extension == FILE_EXTENSION_SC4) {
+		if (!rct1_load_scenario(path))
+			return 0;
+	}
+	else
 		return 0;
```
Comment (Gymnasiast) [inline src/scenario.c]: Could you put braces around this line? It's more consistent with the preceding if and else-if clause.

File: src/windows/loadsave.c [hunk 2]
```diff
@@ -165,7 +165,7 @@ rct_window *window_loadsave_open(int type, char *defaultName)
 	if (!str_is_null_or_empty(defaultName)) {
 		safe_strcpy(_defaultName, defaultName, sizeof(_defaultName));
 	}
-
+
```
Comment (Gymnasiast) [inline src/windows/loadsave.c]: You left a stray tab here.

**Comment by Gymnasiast** [comment]
I left a few formatting nodes, but it otherwise looks good to me code-wise.

**Comment by wolfreak99** [comment]
i did the two updates, but fixup'd them into the commits. i meant to post a seperate commit with the fixes first to show them BEFORE cleaning them up, but got distracted by playing around with git. my bad.

**Pull request merged**